### DISCUSSION TITLE
How to get existing data in queryFn?

### DISCUSSION THREAD

**Discussion by mad-coder-365** [reduxjs/redux-toolkit#3019] [Q&A] [answered]
I have a builder query like
```
getUser: builder.query<IUser | null, null>({
    queryFn: () => ({data: null}),
    async onQueryStarted(
        arg,
        {updateCachedData, queryFulfilled}
    ) {
        const socket = getSocket()
        try {
            socket.emit('getUser', (user) => {
                updateCachedData(() => {
                    return user
                })
            })
            await queryFulfilled
        } catch (err: any) {
            console.log(err)
        }
    },
    keepUnusedDataFor: 30,
    providesTags: ['User']
}),
```
And I call it like 
```
const {data: user} = useGetUserQuery(null, {
    pollingInterval: 60000,
    refetchOnFocus: true,
    refetchOnMountOrArgChange: true
})
```
In the component, I have set it up such that it will show a loading spinner if the user is null. When is component is first loaded it shows a loading spinner then the user data which is expected but after 60 seconds user becomes null and goes back to the loading spinner while it is re-fetching. 

Is it possible to pass existing data inside ```queryFn```? I have tried ```queryFn: (initialData) => ({data: initialData})``` but this doesn't work and it is always null.

How can I update data without making the existing data null using ```queryFn```? Thanks.

**Comment by phryneas** [comment, marked as answer]
Why are you not just doing all of that in `queryFn`?

```js
queryFn: arg => new Promise((resolve, reject) => {
        const socket = getSocket()
        socket.emit('getUser', user => resolve({ data: user }))
 })
```

**Reply by mad-coder-365** [reply]
I am new to this so still figuring out things. I will try your way. Thanks.
